### PR TITLE
Address meson deprecation warnings

### PR DESCRIPTION
**Pull request by evelikov**
The meson build is throwing a couple of deprecation warnings - fix them.

### CODE REVIEW

**Comment by evelikov** [comment]
@sivileri for the win32 fix.

Previously we were using `configure_file()`, which was copying the header early in the build process from `source/va/win32/va_win32.h` to `build/va/va_win32.h`. Now that happens at the very end (the replacement `fs.copyfile()` has different semantics) so the build fails.

File: va/meson.build [hunk 1]
```diff
@@ -318,5 +318,9 @@ if WITH_WIN32
 endif
 
 foreach header : libva_headers_subproject
-  configure_file(output: fs.name(header), input: header, copy: true)
+  if meson.version().version_compare('>= 0.64')
+    fs.copyfile(header)
+  else
+    configure_file(output: fs.name(header), input: header, copy: true)
```
Comment (dvrogozh) [inline va/meson.build:324]: Where I can find information on meson side that it got deprecated? I don't see that here: https://mesonbuild.com/Reference-manual_functions.html#configure_file

File: doc/meson.build [hunk 2]
```diff
@@ -1,4 +1,8 @@
- headerdir = join_paths(meson.source_root(), 'va')
+if meson.version().version_compare('>= 0.56')
+  headerdir = join_paths(meson.project_source_root(), 'va')
+else
+  headerdir = join_paths(meson.source_root(), 'va')
```
Comment (dvrogozh) [inline doc/meson.build:4]: Briefly looking I don't see neither of these functions in meson documentation. And can't check deprecation status. Maybe I just don't know where to look.

**Review by dvrogozh — APPROVED** [review]
looks good

**Comment by evelikov** [comment]
@dvrogozh Is there more I can do here to have this merged?

**Comment by dvrogozh** [comment]
> @dvrogozh Is there more I can do here to have this merged?

Merged.

**Pull request merged**